### PR TITLE
BuildEndpoint refactored, Project and Task API feature files/tests added

### PR DESCRIPTION
**Pull request by rubercuellar**
BuildEndpoint refactored, Project and Task API feature files/tests added

### CODE REVIEW

File: src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/AssertionSteps.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    /**
+     * To get project object by name.
+     *
+     * @param endpoint The endpoint
+     * @param name     The name to search
+     * @return Map<String, ?></>
+     */
+    public Map<String, ?> getObjectByName(final String endpoint, final String name) {
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/AssertionSteps.java]: Do we need this method?

File: src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/RequestSteps.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        return response;
+    }
+
+    /**
+     * To store value.
+     *
+     * @param variableName The variable's name to store
+     */
+    @And("^I stored as \\[([^\"]*)\\]$")
+    public void iStoredAsProject(final String variableName) {
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/RequestSteps.java]: The method name should be generic
not only for project

File: src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/RequestSteps.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import org.fundacionjala.pivotal.cucumber.api.RequestManager;
+import org.fundacionjala.pivotal.cucumber.utils.StoreVariables;
+import org.fundacionjala.pivotal.cucumber.utils.Utils;
+
+/**
+ * Created by Administrator on 6/1/2017.
+ */
+public class RequestSteps {
+
+    private static final List<StoreVariables> STORE_VARIABLES = new ArrayList<>();
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/RequestSteps.java]: Can be this static variable moved to Utils class?

Comment (rubercuellar) [reply src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/RequestSteps.java]: Moved

File: src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/AssertionSteps.java (hunk 1 above)
Comment (rubercuellar) [reply src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/AssertionSteps.java]: Removed

File: src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/RequestSteps.java (hunk 2 above)
Comment (rubercuellar) [reply src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/RequestSteps.java]: Method renamed

File: src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/RequestSteps.java (hunk 3 above)
Comment (rubercuellar) [reply src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/RequestSteps.java]: Moved

File: src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/RequestSteps.java (hunk 2 above)
Comment (caberoDaniel) [reply src/test/java/org/fundacionjala/pivotal/cucumber/stepdefinitions/api/RequestSteps.java]: The method could be made more general for the other components of the project and does not remove

File: src/test/java/org/fundacionjala/pivotal/cucumber/hooks/Hooks.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+  /**
+     * To delete a project.
+     */
+    @After("@deleteProject")
+    public void setup() {
+        for (Iterator<StoreVariables> iter = Utils.getStoreVariables().listIterator(); iter.hasNext();) {
+            StoreVariables var = iter.next();
+            if (var.getName().contains("Project")) {
+                RequestManager.delete(Utils.buildEndpoint("/projects/[" + var.getName() + ".id]"));
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/pivotal/cucumber/hooks/Hooks.java:30]: Don't concatenate Strings using +
Consider using a String.format

File: src/test/java/org/fundacionjala/pivotal/cucumber/hooks/Hooks.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+  /**
+   * To delete workspaces.
+   */
+  @After("@deleteWorkspace")
+  public void resetworkspace() {
+    for (StoreVariables var : Utils.getStoreVariables()) {
+      if (var.getName().contains("Workspace")) {
+        //var.getAttribute("i")
+        RequestManager.delete(Utils.buildEndpoint("/my/workspaces/[" + var.getName() + ".id]"));
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/pivotal/cucumber/hooks/Hooks.java:53]: Don't concatenate Strings using +
Consider using a String.format

File: src/test/java/org/fundacionjala/pivotal/cucumber/hooks/Hooks.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+     */
+    public Hooks() {
+    }
+
+  /**
+     * To delete a project.
+     */
+    @After("@deleteProject")
+    public void setup() {
+        for (Iterator<StoreVariables> iter = Utils.getStoreVariables().listIterator(); iter.hasNext();) {
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/pivotal/cucumber/hooks/Hooks.java:27]: Why are you using an iterator?

Comment (carledriss) [reply src/test/java/org/fundacionjala/pivotal/cucumber/hooks/Hooks.java:27]: never mind
I have just seen all the code.

**Pull request merged**